wetterdienst/dwd/radar/index.py

```diff
@@ -0,0 +1,251 @@
+
+    files_server = list_remote_files(url, recursive=True)
+
+    files_server = pd.DataFrame(
+        files_server, columns=[DWDMetaColumns.FILENAME.value], dtype="str"
+    )
+
+    # Some directories have both "---bin" and "---bufr" files within the same directory,
+    # so we need to filter here by designated RadarDataFormat. Example:
+    # https://opendata.dwd.de/weather/radar/sites/px/boo/
+    if format is not None:
+        if format == RadarDataFormat.BINARY:
+            files_server = files_server[
+                files_server[DWDMetaColumns.FILENAME.value].str.contains("--bin")
+            ]
+        elif format == RadarDataFormat.BUFR:
+            files_server = files_server[
+                files_server[DWDMetaColumns.FILENAME.value].str.contains("--buf")
+            ]
+
+    # Decode datetime of file for filtering.
+    if parse_datetime:
+
+        files_server[DWDMetaColumns.DATETIME.value] = files_server[
+            DWDMetaColumns.FILENAME.value
+        ].apply(get_date_from_filename)
+
+        files_server = files_server.dropna()
+
+    return files_server
+
+
+@fileindex_cache_five_minutes.cache_on_arguments()
+def create_fileindex_radolan_cdc(
+    time_resolution: TimeResolution, period_type: PeriodType
+) -> pd.DataFrame:
+    """
+    Function used to create a file index for the RADOLAN_CDC product. The file index
+    will include both recent as well as historical files. A datetime column is created
+    from the filenames which contain some datetime formats. This datetime column is
+    required for later filtering for the requested file.
+
+    :param time_resolution: Time resolution for RadarParameter.RADOLAN_CDC,
+                            either daily or hourly or 5 minutes.
+    :param period_type:     Period type for RadarParameter.RADOLAN_CDC
+
+    :return:                File index as DataFrame
+    """
+    file_index = create_fileindex_radar(
+        parameter=RadarParameter.RADOLAN_CDC,
+        time_resolution=time_resolution,
+        period_type=period_type,
+    )
+
+    file_index = file_index[
+        file_index[DWDMetaColumns.FILENAME.value].str.contains("/bin/")
+        & file_index[DWDMetaColumns.FILENAME.value].str.endswith(
+            (ArchiveFormat.GZ.value, ArchiveFormat.TAR_GZ.value)
+        )
+    ]
+
+    # Decode datetime of file for filtering.
+    file_index[DWDMetaColumns.DATETIME.value] = file_index[
+        DWDMetaColumns.FILENAME.value
+    ].apply(
+        lambda filename: parse(
+            RADOLAN_DT_PATTERN.findall(filename)[0],
+            date_formats=[DatetimeFormat.YM.value, DatetimeFormat.ymdhm.value],
+        )
+    )
+
+    return file_index
+
+
+def build_path_to_parameter(
+    parameter: RadarParameter,
+    site: Optional[RadarSite] = None,
+    format: Optional[RadarDataFormat] = None,
+    subset: Optional[RadarDataSubset] = None,
+    time_resolution: Optional[TimeResolution] = None,
+    period_type: Optional[PeriodType] = None,
+) -> str:
```

Comment:
Maybe return Path objects here?

Reply:
Using Path objects for URLs has caused too much trouble in the past.

Reply:
Okay. Fine for me